plugins/scaffolder/src/components/TemplateList/TemplateList.tsx

```diff
@@ -32,41 +35,84 @@ export type TemplateListProps = {
   TemplateCardComponent?:
     | ComponentType<{ template: TemplateEntityV1beta2 }>
     | undefined;
+  swimlane?: {
+    title: React.ReactNode;
+    filter: (entity: Entity) => boolean;
+  };
 };
 
-export const TemplateList = ({ TemplateCardComponent }: TemplateListProps) => {
+type TemplateCardGridProps = {
+  Card: ComponentType<{ template: TemplateEntityV1beta2 }>;
+  entities: Entity[];
+};
+const TemplateCardGrid = ({ Card, entities }: TemplateCardGridProps) => {
+  return (
+    <ItemCardGrid>
+      {entities &&
+        entities?.length > 0 &&
+        entities.map((template: Entity) => (
+          <Card
+            key={stringifyEntityRef(template)}
+            template={template as TemplateEntityV1beta2}
+          />
+        ))}
+    </ItemCardGrid>
+  );
+};
+
+export const TemplateList = ({
+  TemplateCardComponent,
+  swimlane,
+}: TemplateListProps) => {
   const { loading, error, entities } = useEntityListProvider();
   const Card = TemplateCardComponent || TemplateCard;
+  if (swimlane === null || swimlane === undefined) {
```

Comment:
Hmm, do you think this would be cleaner just to resolve `filteredEntities` and the title ahead of time, and have the same display logic for both situations?

```tsx
const maybeFilteredEntities = swimlane ? entities.filter(e => swimlane.filter(e)) : entities;
const title = swimlane ? {swimlane.title} : <ContentHeader title="All Templates" />;

if (swimlane && maybeFilteredEntities.length === 0) {
  return null;
}

return (
  // Shared TSX
);
```

Reply:
Yeah definitely, can update!